### PR TITLE
T1041 - SDS 1

### PR DESCRIPTION
**Pull request by langjer**
- Correction of the retrieval of information about the reception center.
- Adjustment of the layout of the childpack for Italian information on one line.
- Implementation of data control, retrieves information as needed.

### CODE REVIEW

File: child_compassion/wizards/print_childpack.py [hunk 1]
```diff
@@ -56,16 +56,34 @@ def _default_lang(self):
...
         )
+        lang = self.lang
+        lang_code = lang[:2]
+        lang_map = self.env["compassion.project.description"]._supported_languages()
+        desc_field = 'desc_' + lang_code
+
+        for child in children:
+            if not getattr(child.project_id, lang_map.get(lang)):
+                if child.project_id.update_informations():
+                    child.project_id = self.env["compassion.project"].search([('id', '=', child.project_id.id)])
```
Comment (ecino) [inline child_compassion/wizards/print_childpack.py]:
```suggestion
                child.project_id.update_informations()
```

File: child_compassion/wizards/print_childpack.py [hunk 2]
```diff
@@ -56,16 +56,34 @@ def _default_lang(self):
...
             .browse(self.env.context.get("active_ids"))
             .with_context(lang=self.lang)
         )
+        lang = self.lang
+        lang_code = lang[:2]
+        lang_map = self.env["compassion.project.description"]._supported_languages()
+        desc_field = 'desc_' + lang_code
+
+        for child in children:
+            if not getattr(child.project_id, lang_map.get(lang)):
```
Comment (ecino) [inline child_compassion/wizards/print_childpack.py]:
```suggestion
            if not hasattr(child.project_id, lang_map.get(lang)):
```

File: child_compassion/wizards/print_childpack.py [hunk 3]
```diff
@@ -56,16 +56,34 @@ def _default_lang(self):
...
+                if child.project_id.update_informations():
+                    child.project_id = self.env["compassion.project"].search([('id', '=', child.project_id.id)])
+            if not getattr(child, desc_field):
+                child.project_id.get_info()
+                if child.get_infos():
+                    children[child] = self.env["compassion.child"].search([('id', '=', child.id)])
```
Comment (ecino) [inline child_compassion/wizards/print_childpack.py]:
```suggestion
            if not hasattr(child, desc_field):
                child.project_id.get_info()
                child.get_infos()
```

**Review by ecino — CHANGES_REQUESTED** [review]

File: child_compassion/wizards/print_childpack.py (hunk 2 above)
Comment (langjer) [reply child_compassion/wizards/print_childpack.py]: hasattr retrun True in all case

File: child_compassion/wizards/print_childpack.py (hunk 1 above)
Comment (langjer) [reply child_compassion/wizards/print_childpack.py]: This change has been applied in the last commit

File: child_compassion/wizards/print_childpack.py (hunk 3 above)
Comment (langjer) [reply child_compassion/wizards/print_childpack.py]: - hasattr retrun True in all case

**Pull request merged**